components/camel-cmis/src/main/java/org/apache/camel/component/cmis/CMISProducer.java

```diff
@@ -113,6 +112,226 @@ private CmisObject createNode(Exchange exchange) throws Exception {
         }
     }
 
+    private List<String> deleteFolder(Exchange exchange) throws Exception {
+        validateRequiredHeader(exchange, PropertyIds.OBJECT_ID);
+
+        Message message = exchange.getIn();
+
+        String objectId = message.getHeader(PropertyIds.OBJECT_ID, String.class);
+        Folder folder = (Folder) getSessionFacade().getObjectById(objectId);
+        return folder.deleteTree(true, UnfileObject.DELETE, true);
+    }
+
+    private void deleteDocument(Exchange exchange) throws Exception {
+        validateRequiredHeader(exchange, PropertyIds.OBJECT_ID);
+
+        Message message = exchange.getIn();
+
+        String objectId = message.getHeader(PropertyIds.OBJECT_ID, String.class);
+        Document document = (Document) getSessionFacade().getObjectById(objectId);
+
+        document.deleteAllVersions();
+    }
+
+    private void moveDocument(Exchange exchange) throws Exception
+    {
+        validateRequiredHeader(exchange, CamelCMISConstants.CMIS_DESTIONATION_FOLDER_ID);
+        validateRequiredHeader(exchange, CamelCMISConstants.CMIS_SOURCE_FOLDER_ID);
+        validateRequiredHeader(exchange, CamelCMISConstants.CMIS_OBJECT_ID);
+
+        Message message = exchange.getIn();
+
+        String destinationFolderId = message.getHeader(CamelCMISConstants.CMIS_DESTIONATION_FOLDER_ID, String.class);
+        String sourceFolderId = message.getHeader(CamelCMISConstants.CMIS_SOURCE_FOLDER_ID, String.class);
+        String objectId = message.getHeader(CamelCMISConstants.CMIS_OBJECT_ID, String.class);
+
+
+        Folder sourceFolder = (Folder) getSessionFacade().getObjectById(sourceFolderId);
+        Folder targetFolder = (Folder) getSessionFacade().getObjectById(destinationFolderId);
+
+        Document document = (Document) getSessionFacade().getObjectById(objectId);
+
+        if(document != null)
+        {
+            if(document.getAllowableActions().getAllowableActions().contains(Action.CAN_MOVE_OBJECT) == false)
+            {
+                throw new CmisUnauthorizedException("Current user does not have permission to move " + objectId + document.getName());
```

Comment:
Usually, we don't use third party exceptions in components. You can choose another one and see example from other components.